packages/quantic/cypress/e2e/default-2/search-box/search-box-expectations.ts

```diff
@@ -0,0 +1,142 @@
+import {InterceptAliases} from '../../../page-objects/search';
+import {should} from '../../common-selectors';
+import {AriaLiveExpectations} from '../../default-1/aria-live/aria-live-expectations';
+import {SearchBoxSelectors, SearchBoxSelector} from './search-box-selectors';
+
+function searchBoxExpectations(selector: SearchBoxSelector) {
+  return {
+    displayClearRecentQueriesButton: (display: boolean) => {
+      selector
+        .clearRecentQueriesButton()
+        .should(display ? 'exist' : 'not.exist')
+        .logDetail(
+          `${should(display)} display the 'clear recent queries' button`
+        );
+    },
+    querySuggestionsEquals: (querySuggestions: String[]) => {
+      querySuggestions.forEach((querySuggestion, index) => {
+        selector
+          .querySuggestion(index)
+          .invoke('attr', 'data-rawvalue')
+          .should('equal', querySuggestion)
+          .logDetail(
+            'should display the query suggestions properly and in the correct order'
+          );
+      });
+    },
+    searchWithQuery: (
```

Comment:
Indeed that's true.

Reply:
What kind of timing issues would you expect?

The localStorage should "reset" whenever we want it to reset? for example if you had a 

```
search(query)
...
intercept, validate property `q`;
...
validate localStorage
```
If you make a new test after that that navigates back to the page, the browser context shouldn't be kept AFAIK.

Reply:
Anyway it's not a big deal, but I'm curious why it's so timing sensitive that we have to put them together.

Reply:
The thing is that the Cypress commands are asynchronous, so when you do
```
Expect.myFirstExpectation() // Executes a cypress command to validate something
Expect.mySecondExpectation() // Executes a cypress command to validate something else
```
it doesn't necessarily mean that `myFirstExpectation` will be executed before  `mySecondExpectation`.

in our case, we first start by waiting for the request, then using the `.then()` we validate that the search query was made with the expected parameters and we confirm that the local storage was updated successfully.

if the local storage validation was in another command, it means that, because cypress commands are asynchronous, the validation of local storage could happen before that the request was made which will make the test fail.